Sming/Components/Network/src/Network/TcpConnection.cpp

```diff
@@ -423,6 +423,29 @@ err_t TcpConnection::internalOnConnected(err_t err)
 	return res;
 }
 
+bool TcpConnection::enableSsl(const String& hostName)
+{
+	if(tcp == nullptr) {
+		return false;
+	}
+
+	if(tcp->state != ESTABLISHED) {
+		return false;
+	}
+
+	if(!sslCreateSession()) {
+		return false;
+	}
+
+	if(hostName) {
+		ssl->hostName = hostName;
+	}
+
+	useSsl = true;
+	useSsl = (internalOnConnected(ERR_OK) == ERR_OK);
```

Comment:
Remove redundant `useSsl = true;`

Reply:
I know that the code looks a bit wierd but in `internalOnConnected()`  useSsl has to be true to kick start the SSL connection.

```C++
err_t TcpConnection::internalOnConnected(err_t err)
{
	debug_tcp_d("connected: useSSL: %d, Error: %d", useSsl, err);

	if(useSsl && err == ERR_OK) {
```

Reply:
May I suggest a comment and/or slight rewrite:

```
useSsl = true;
if (internalOnConnected(ERR_OK) != ERR_OK) {
  useSsl = false;
}
```

Reply:
> May I suggest a comment and/or slight rewrite:

Sure. I will apply your change in a minute.